ci: build and push Docker images on release

vayan opened the pull request:
Also refactor the "publish to pypi" workflow into a global "release" workflow.
because the images depend on the package being published.

Run sample: 

<a href="https://github.com/ccpgames/protoplasm/actions/runs/8732653645"><img width="1333" alt="Screenshot 2024-04-18 at 13 08 29" src="https://github.com/ccpgames/protoplasm/assets/2945291/2f008f29-ef4b-4b5a-9c6d-be010f7fe8d3"></a>

CCP-Zeulix commented:
Sheer brilliance!

Looks really good and if it works, feel free to "ship-it" :D

The only thing that might be worth looking into is to check if it's possible in the pipelines to check if the release was flagged as a "pre-release" and if so, skip the image building part (or make it optional and triggered by manual button pressing for pre-releases). We basically only want images with stable versions as they're used as base images for production services.

vayan commented:
Oh you're right! Added a "prereleased" workflow that will only publish to pypi and updated the previous one (pypi+images)  to only trigger on "released" events.
